src/main/java/com/hedera/hashgraph/sdk/IdUtil.java

```diff
@@ -0,0 +1,17 @@
+package com.hedera.hashgraph.sdk;
+
+import com.google.common.base.Splitter;
+
+public final class IdUtil {
+
+    private IdUtil() {
+    }
+
+    public static long[] parseIdString(String id) {
+        var rawNums = Splitter.on('.')
+            .split(id)
+            .iterator();
+
+        return new long[]{Long.parseLong(rawNums.next()), Long.parseLong(rawNums.next()), Long.parseLong(rawNums.next())};
+    }
```

Comment:
Instead of returning an array and indexing into that in the constructors of the other types, why not have something like this:

```java
@FunctionalInterface
public interface WithIdNums<R> {
    R apply(long shardNum, long realmNum, long entityNum);
}
```

And then this becomes: 

```java
public static <R> R parseIdString(String id, WithIdNums<R> withIdNums) {
    var rawNums = Splitter.on('.')
        .split(id)
        .iterator();

    return withIdNums.apply(Long.parseLong(rawNums.next()), Long.parseLong(rawNums.next()), Long.parseLong(rawNums.next()));
}
```

And invocation becomes as simple as `IdUtils.parseIdString(idString, AccountId::new)`, etc.